### PR TITLE
RavenDB-16620: prevent possible double dispose of return context

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs [hunk 1]
```diff
@@ -269,7 +270,12 @@ private async Task<Operation> ImportInternalAsync(DatabaseSmugglerImportOptions
             }
             catch (Exception e)
             {
-                returnContext?.Dispose();
+                if (requestTask == null)
+                {
+                    // handle the possible double dispose of return context
+                    returnContext?.Dispose();
```
Comment (ayende) [inline src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs:289]: Double dispose should always be safe, no?

Comment (karmeli87) [reply src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs:289]: double dispose of the context is safe, but not the dispose of the `ReturnRequestContext`, even if the dispose is not concurrent (it isn't thread safe) you might hit different dispose path if the low mem event going on and off

Comment (ayende) [reply src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs:289]: There should be no case that the dispose here is concurrent. 
And the `ReturnRequestContext` should be made for repeated calls.

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs [hunk 2]
```diff
@@ -127,6 +133,13 @@ private async Task<Operation> ExportAsync(DatabaseSmugglerExportOptions options,
                     getOperationIdCommand.NodeTag,
                     additionalTask);
             }
+            catch (Exception e)
+            {
+                if (requestTask == null)
+                    returnContext?.Dispose();
+
+                throw e.ExtractSingleInnerException();
```
Comment (ayende) [inline src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs:141]: Why are you throwing it like this? We are losing the stack trace in this manner.

Comment (ayende) [inline src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs:141]:
```suggestion
                throw;
```

Comment (ppekrol) [reply src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs:141]: It is the same in every place, probably to avoid throwing AggregateException

**Pull request merged**